cli/internal/cmd/add/add.go

```diff
@@ -0,0 +1,161 @@
+package add
+
+import (
+	"fmt"
+	"html/template"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/aerogear/charmil/cli/internal/common/modname"
+	"github.com/aerogear/charmil/cli/internal/factory"
+	"github.com/aerogear/charmil/cli/internal/template/add"
+	"github.com/aerogear/charmil/core/color"
+	"github.com/spf13/cobra"
+)
+
+// TemplateData defines fields that will store all the data used for generating templates
+type TemplateData struct {
+	// Stores value of the `CmdPath` local flag. Default Value: "."
+	CmdPath string
+
+	// Stores value of the `CmdName` local flag
+	CmdName string
+
+	// Stores the name of the root module (extracted from go.mod file)
+	ModName string
+}
+
+// Initializes a zero-valued struct
+var tmplData = TemplateData{}
+
+func AddCommand(f *factory.Factory) (*cobra.Command, error) {
+	cmd := &cobra.Command{
+		Use:           f.Localizer.LocalizeByID("add.cmd.use"),
+		Short:         f.Localizer.LocalizeByID("add.cmd.short"),
+		Long:          f.Localizer.LocalizeByID("add.cmd.long"),
+		Example:       f.Localizer.LocalizeByID("add.cmd.example"),
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Extracts the module name from `go.mod` file and stores it
+			modName, err := modname.GetModuleName()
+			if err != nil {
+				return err
+			}
+			tmplData.ModName = modName
+
+			if err := generateCommand(); err != nil {
+				return err
+			}
+
+			f.Logger.Infof(color.Success("%s command has been created in %s directory\n"), tmplData.CmdName, tmplData.CmdPath+"/"+tmplData.CmdName)
+
+			return nil
+
+		},
+	}
+
+	// Adds local flags
+	cmd.Flags().StringVarP(&tmplData.CmdPath, f.Localizer.LocalizeByID("add.flag.cmdPath.name"), "c", ".", f.Localizer.LocalizeByID("add.flag.cmdPath.description"))
+	cmd.Flags().StringVarP(&tmplData.CmdName, f.Localizer.LocalizeByID("add.flag.cmdName.name"), "s", "", f.Localizer.LocalizeByID("add.flag.cmdName.description"))
```

Comment:
Maybe you can add a flag here that takes a target path for the language file and then generate the file in that location.
The crud command does this too. Check [this out](https://github.com/aerogear/charmil/blob/f4f743e2f55669cc3a849521deb37515d2bef8b8/cli/internal/cmd/crud/crud.go#L62) for reference.

Reply:
I think, we can leave it? it would be difficult for the user to give two paths. It's just a file, user can move it wherever they wants manually.

it will be better to generate the locales file with the command file. to let the user figure out what really happened.